jwplatform/upload.py

```diff
@@ -0,0 +1,137 @@
+import logging
+from enum import Enum
+from hashlib import md5
+import requests
+from requests import HTTPError
+
+from jwplatform import constants, common
+from jwplatform.upload_errors import DataIntegrityError, MaxRetriesExceededError
+
+
+class UploadType(Enum):
+    direct = "direct"
+    multipart = "multipart"
+
+class MultipartUpload:
+
+    def __init__(self, client, upload_id: str, file, target_part_size, retry_count):
+        self.upload_id = upload_id
```

Comment:
I would recommend making these attributes and the ones on SingleUpload pseudo-private (prefixed with underscore) unless there's a case where we predict a user needing to access them. It's easy to add support later if someone asks for it but harder to take something away if we want to change some underlying functionality here.